src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -587,20 +616,61 @@ impl AuthProvider for OpenIDConnectAuthProvider {
                 // see if we can decode, decrypt and deserialize the users token
                 // into a login session structure
                 let session = self.session_cache.decode(token, &self.session_key)?;
-
                 let status = session.status();
 
-                let mut new_auth = None;
-                if status != SessionStatus::Active {
-                    new_auth = self.try_refresh_token(&session);
-                    if new_auth.is_none() && status == SessionStatus::Expired {
-                        return Err(Error::ApiInvalidCredentials(
-                            "Session expired, please login again".to_string(),
-                        ));
-                    }
+                // Token found in cache and active; all good, do an early return
+                if status == SessionStatus::Active {
+                    return Ok(Some(ActorDef::user(session.id, session.attributes, None)));
                 }
 
-                Ok(Some(ActorDef::user(session.id, session.attributes, new_auth)))
+                let new_auth = match self.try_refresh_token(&session) {
+                    Ok(auth) => {
+                        trace!(
+                            "OpenID Connect: Succesfully refreshed token for user \"{}\"",
+                            &session.id
+                        );
+                        auth
+                    }
+                    Err(err) => {
+                        trace!("OpenID Connect: rfc-6749 5.2 Error Response returned...");
```

Comment:
I'm curious about the "don't shout rfc" commit you made that lowercased "RFC" in these log statements.

This change of case makes these log statements inconsistent with other log statements in this file that use uppercase for RFC. If you look at https://www.ietf.org/standards/rfcs/ you'll see that RFC is rendered in capital letters, in English at least acronyms are always in all capitals, and Googling a bit I found [this](https://tools.ietf.org/html/draft-carpenter-rfc-citation-recs-01#section-5.2) which also states that RFCs should be referenced in uppercase:

> The correct typography includes a blank space.  Both "RFC2026" and "RFC-2026" are incorrect as a descriptive term.

Although it also states that using a dash is incorrect so in that case the existing log comments are also incorrect...

Reply:
I find a mixture of lowercase and uppercase harder to read. Not a big deal, so if the standard is RFC, RFC it is